pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -34,9 +34,13 @@ type Constraints struct {
 	Taints Taints `json:"taints,omitempty"`
 	// Requirements are layered with Labels and applied to every node.
 	Requirements Requirements `json:"requirements,inline,omitempty"`
-	// KubeletConfiguration are options passed to the kubelet when provisioning nodes
+	// KubeletConfiguration are options passed to the kubelet when provisioning nodes, this
+	// is a subset of https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration.
 	//+optional
 	KubeletConfiguration KubeletConfiguration `json:"kubeletConfiguration,omitempty"`
+	// ContainerRuntimeConfiguration are options passed to the container runtime when provisioning nodes
+	//+optional
+	ContainerRuntimeConfiguration ContainerRuntimeConfiguration `json:"containerRuntimeConfiguration,omitempty"`
```

Comment:
I'm not comfortable including this custom spec into our API surface. Does this block your use case?

Reply:
Yes, please see https://github.com/aws/karpenter/pull/1346#discussion_r807234988 (above)

Reply:
I hear you on this request. Still -- this is API surface and we need to get it right.

Can we at least remove from this PR so that we can move forward with the kubeletconfiguration changes which are less contentious? We can address the remaining gap in a followon.